core/src/main/java/com/epam/esm/dao/GiftDao.java

```
@@ -0,0 +1,20 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.exception.DaoException;
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+
+import java.util.List;
+
+public interface GiftDao {
+    List<GiftCertificateEntity> findAllGifts();
+    GiftCertificateEntity findGiftById(Long giftId) throws DaoException;
```

if exception is not checked it is better to remove it from interface